hoodie-common/src/main/java/com/uber/hoodie/common/model/FileSlice.java

```diff
@@ -81,10 +99,26 @@ public String getFileId() {
     return Optional.ofNullable(dataFile);
   }
 
+  public Optional<String> getOutstandingCompactionInstant() {
```

Comment:
if the getter is Optional, probably make the variable also Optional<String> ?

Reply:
Optional is not serializable. Hence, the member variable is not marked Optional

Reply:
Is there a way we can add this to Kryo serializer in HoodieWriteClient ?

  public static SparkConf registerClasses(SparkConf conf) {
    conf.registerKryoClasses(
        new Class[] {HoodieWriteConfig.class, HoodieRecord.class, HoodieKey.class, Optional.class});
    return conf;
  }

Reply:
I haven't tried it. I thought we need to support use-cases where OSS ingestion clients use  default spark serialization ? Is it not the case ?